### ISSUE TITLE
run hr_res('train')  has error

### ISSUE THREAD

**Issue by tzhang2014** [peiyunh/tiny#33]
hello,dear peiyun ,
I can run the demo ,but when I run the hr_res('train') got the follow error:

Start using dagnn.DetLoss for loss
cnn_train_dag_hardmine: resetting GPU
train: epoch 01: 1/1074:Error using vl_nnconv
Out of memory on device. To view more detail about available memory on the GPU, use 'gpuDevice()'. If the problem persists, reset the GPU by calling
'gpuDevice(1)'.

Error in dagnn.Conv/forward (line 11)
outputs{1} = vl_nnconv(...

Error in dagnn.Layer/forwardAdvanced (line 85)
outputs = obj.forward(inputs, {net.params(par).value}) ;

Error in dagnn.DagNN/forward (line 28)
obj.layers(l).block.forwardAdvanced(obj.layers(l)) ;

Error in cnn_train_dag_hardmine>process_epoch (line 286)
net.forward(inputs, opts.derOutputs);

Error in cnn_train_dag_hardmine (line 148)
[stats.train(epoch),prof] = process_epoch(net, state, opts, 'train') ;

Error in cnn_widerface (line 212)
[net, info] = trainFn(net, imdb, getBatchFn(batchGetter, opts, net.meta), ...

Error in hr_res101 (line 42)
cnn_widerface('inputSize', inputSize, ...

compile_mex was successfully, can you help me solve it ? Thx

**Comment by zhangjinsong3** [comment]
out of memory!
train to change the batchSize=4 ,for example : batchSzie =1.
Because the resnet101 need much more memory.

**Comment by tzhang2014** [comment]
thank you ! I have set opts.batchSize = 10  to opts.batchSize = 1in cnn_widerface.m and opts.batchSize = 256  to opts.batchSize = 1 in cnn_train_dag_hardmine.m ,but remain get the same error ,

**Comment by peiyunh** [comment]
Parameters set in `hr_res101.m` will override default ones in both `cnn_widerface.m` and `cnn_train_dag_hardmine.m`.

**Comment by FredaZhang338** [comment]
hi  dear @peiyunh I haved compiled and tested as you said,and I can run the demo,but when I run hr_res101('train');I got the errors as floows:
( I have changed the MAX_INPUT_DIM in hr_res101.m to  1500)
Lab 1: 
  train: epoch 01:   1/269:
Lab 2: 
  train: epoch 01:   1/269:
Lab 3: 
  train: epoch 01:   1/269:
Lab 4: 
  train: epoch 01:   1/269:
Error using dagnn.Conv/forward (line 11)
Error detected on worker 1.
Out of memory on device. To view more detail about available memory on the GPU, use 'gpuDevice()'.
If the problem persists, reset the GPU by calling 'gpuDevice(1)'.

Error in dagnn.Layer/forwardAdvanced (line 85)
      outputs = obj.forward(inputs, {net.params(par).value}) ;

Error in dagnn.DagNN/forward (line 28)
  obj.layers(l).block.forwardAdvanced(obj.layers(l)) ;

Error in cnn_train_dag_hardmine>process_epoch (line 286)
        net.forward(inputs, opts.derOutputs);

Error in cnn_train_dag_hardmine>(spmd body) (line 158)
      [stats_.train, prof_] = process_epoch(net_, state, opts, 'train') ;

Error in cnn_train_dag_hardmine>(spmd) (line 156)
    spmd

Error in cnn_train_dag_hardmine (line 156)
    spmd

Error in cnn_widerface (line 212)
[net, info] = trainFn(net, imdb, getBatchFn(batchGetter, opts, net.meta), ...

Error in hr_res101 (line 41)
    cnn_widerface('inputSize', inputSize, ...


I have tried to reset the gpuDevice and tried to change the batchsize smaller ,but another error came out.So I am wondering if you have any idea to fix this problem. thank you so much!
